cookbook/session/locale_sticky_session.rst

```diff
@@ -106,3 +106,86 @@ method::
     {
         $locale = $request->getLocale();
     }
+
+Setting the locale based on the user entity
+-------------------------------------------
+
+You might want to improve even further and want to define the locale based on
+the user entity of the logged in user. However since the `LocaleListener` is called
+before the `FirewallListener`, which is responsible for handling authentication and
+is setting the user token into the `TokenStorage`, you have no access to the user
+which is logged in.
+
+First lets pretend you have defined a property locale in your User Entity which you
+want to be used as the locale for the given user. In order to achieve the wanted locale
```

Comment:
I find this phrase too long and a bit confusing. Maybe we should reword it or split it:

```
In order to achieve the wanted locale configuration you can set the locale
which is defined for the user to the session right after the login.
```